src/Microsoft.ML.HalLearners/OlsLinearRegression.cs

```diff
@@ -106,7 +106,7 @@ internal OlsLinearRegressionTrainer(IHostEnvironment env, Arguments args)
             advancedSettings?.Invoke(args);
             args.FeatureColumn = featureColumn;
             args.LabelColumn = labelColumn;
-            args.WeightColumn = weightColumn;
+            args.WeightColumn = weightColumn != null ? Optional<string>.Explicit(weightColumn) : Optional<string>.Implicit(DefaultColumnNames.Weight);
```

Comment:
>weightColumn != null  [](start = 32, length = 21)

so should i make a similar fix for SDCA ?

Reply:
than we'd need a separate PR to fix the ctor in KMeans .. and this PR does take care of both.. 

---
In reply to: [248117752](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248117752) [](ancestors = 248117752)

Reply:
my concern is by coupling the weights changes   with API changes, we are  making ourselves susceptible to inconsistency across PRs etc.

So I am not sure at this point if I need to make similar changes for the  weights bug in my existing PRs  *or not*

Would be much more convenient if we have consistent story for weights, get that reviewed + checked in and then  move on with API  fixing 

---
In reply to: [248158333](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248158333) [](ancestors = 248158333,248117752)